### ISSUE TITLE
Inconsistent results using your checkpoint

### ISSUE THREAD

**Issue by d12306** [asharakeh/probdet#2]
Hi, @asharakeh , thanks for your implementation, actually I tried to reproduce the results of the faster-rcnn with energy score for inference, and I get inconsistent results with table.1 about the number of TP, local errors, and duplicates. The calibration error is also different, please see the attached figure for my results. I tested the faster_rcnn_R_50_FPN_3x_reg_var_es.pth on COCO validation set.
![attach](https://user-images.githubusercontent.com/42554178/109286754-a4f8ea00-785d-11eb-8bfc-1efc0c7810ce.PNG)
 
And I use this command line to run the code.
`
 python src/apply_net.py  --dataset-dir  /nobackup-slow/dataset/my_xfdu/coco2017  --test-dataset coco_2017_custom_val --config-file COCO-Detection/faster-rcnn/faster_rcnn_R_50_FPN_3x_reg_var_es.yaml  --inference-config Inference/standard_nms.yaml --random-seed 0 --image-corruption-level 0
`

Are there any mistakes I make?

Thanks,

**Comment by d12306** [comment]
I guess it has something to do with the mAP_res.txt, but I do not know how to generate such a txt

**Comment by asharakeh** [comment]
@d12306 Yes, you absolutely need to generate mAP_res..txt first as it contains the classification threshold used to filter out detections. What you are doing now is evaluating every single detection output from faster RCNN, which produces 100 detections per image.

To generate mAP_res.txt, as well as all other evaluations, please run apply_net.py with the flag --eval-only.

**Comment by asharakeh** [comment]
@d12306 Thank you for pointing this out! I just reviewed the apply_net.py script and there is a commented line that should not be commented. 

It should now be fixed! Please try the script again, mAP_res.txt should now work properly.

**Comment by d12306** [comment]
@asharakeh ,thanks!

**Comment by d12306** [comment]
@asharakeh , hi, sorry for another question, 

I use the checkpoint of deterministic faster rcnn and test it on coco val 2017, the result seems to not match the result in the detectron2 model zoo (37.0 compared to 40.2),  there is no different between the provided config compared to the official one in detectron2, except for the learning rate, but you decrease the batch size for 4x. so that shouldn't be a problem.

Do you change any other hyperparameters?

Thanks,

**Comment by asharakeh** [comment]
@d12306 No worries! No, all remaining hyperparameters are kept the same. The 3% increase in mAP might be from the reduction of the learning rate, or from the difference in hardware used for training.

Basically, I got ~37% on 5 differently initialized networks (used for ensembles), meaning that the 40.2% in the model zoo is due to a larger number of GPUs where one can use larger batch sizes and higher learning rates.
